website/docs/feature-flag-tutorials/python/implementing-feature-flags.md

```diff
@@ -0,0 +1,346 @@
+
+![A screenshot of the Flask Surveys Container App](/img/python-tutorial-survey-example.png)
+
+In this tutorial, you will learn how to set up and use Python feature flags with Unleash. We will use the Flask Surveys container app to implement the feature flag solution, which will roll out a feature that allows users to delete surveys they create.
+
+At the end of this tutorial, you will be able to turn on the feature flag and activate a route that will remove surveys from the database.
+
+Here are the steps we will cover in this tutorial:
+1. [Feature flag best practices for back-end applications](#1-unleash-best-practice-for-backend-apps)
+2. [Spin up a local flag provider](#2-install-a-local-feature-flag-provider)
+3. [Configure a feature flag](#3-create-and-configure-the-feature-flag)
+4. [Add Unleash to a Python Flask app](#4-add-unleash-to-a-python-app)
+5. [Toggle the database deletion route](#5-use-a-feature-flag-to-roll-out-a-delete-method)
+6. [Verify the toggle experience](#6-verify-the-toggle-experience)
+7. [Improve feature flag error handling](#7-improve-a-feature-flag-implementation-with-error-handling)
+
+
+## Prerequisites
+
+
+In this tutorial, you will need the following:
+- A web browser like Chrome or Firefox
+- Git
+- Docker
+- (Optional) a code editor like Visual Studio Code
+
+
+![An architectural diagram of our Python app using Unleash feature flags](/img/python-flask-unleash-architecture.png)
+
+
+This architecture diagram breaks down how the Python app works with Unleash to control feature flags.
+
+Your Python app is an **Evaluation Service**, which is meant to evaluate feature flags and pass evaluated results down to the SDK in the client application. In this context, the results will be passed to the Python SDK. 
+
+The Unleash Server is a **Feature Flag Control Service**, which is a service that manages your feature flags and is used to retrieve flag data from (and send data to, especially when not using a UI). The Unleash server has a UI for creating and managing projects and feature flags. There are also [API commands available in our documentation](https://docs.getunleash.io/reference/api/unleash) to perform the same actions straight from your CLI or server-side app.
+
+An Evaluation Service and a Feature Flag Control Service work hand in hand for you to leverage feature flags. For this tutorial, we will be using both to toggle extended functionality for surveys hosted in the Python app database.
+
+
+## 1. Unleash best practice for backend apps
```

Comment:
This section feels a bit awkward to me. I'm not sure it fits well here. Everything else seems to be pretty concrete advice targeted towards beginners using Unleash. This is largely about scaling feature flag systems from an expert perspective. These are definitely things that someone getting a crash course in Unleash doesn't need to know. If you want this stuff, personally add it as a further reading section or addendum

Reply:
I could make it a note instead and define the terms evaluation service and FF control service, but yes we don't need too much detail in this tutorial if we want the reader up and running quickly.

Reply:
Ball is in your court! Just an opinion of mine, you're the Word Smith!